fotoapparat/src/main/java/io/fotoapparat/hardware/orientation/OrientationSensor.java

```diff
@@ -38,12 +40,15 @@ public void stop() {
     }
 
     @Override
-    public void onRotationChanged() {
+    public void onRotationChanged(int newOrientation) {
         if (listener != null) {
-            int rotation = screenOrientationProvider.getScreenRotation();
-            if (rotation != lastKnownRotation) {
-                listener.onOrientationChanged(rotation);
+            // reduce orientation to multiple of 90
+            int orientation = ((newOrientation + 45) / 90 * 90) % 360;
+            int rotation = this.screenOrientationProvider.getScreenRotation();
```

Comment:
Given that we'll remove `lastKnownRotation` let's move it inside of the `if` block so that this call will happen less often.